### PR TITLE
MAKE-925: add lint tests to jasper suite

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: cmd/run-linter/run-linter.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+// String prints the results of a linter run in gotest format.
+func (r *result) String() string {
+	buf := &bytes.Buffer{}
+
+	fmt.Fprintln(buf, "=== RUN", r.name)
+	if r.passed {
+		fmt.Fprintf(buf, "--- PASS: %s (%s)", r.name, r.duration)
+	} else {
+		// fmt.Fprintln(buf, "    CMD:", r.cmd)
```
Comment (Kimchelly) [inline cmd/run-linter/run-linter.go]: Remove

File: harness.poplar [hunk 2]
```diff
@@ -1,5 +1,4 @@
 // +build none
```
Comment (Kimchelly) [inline harness.poplar]: Why were these files renamed to poplar? What does the poplar extension do?

File: job.go [hunk 3]
```diff
@@ -221,13 +253,22 @@ func amboyForegroundOutputJobFactory(pc ProcessConstructor) *amboyForegroundOutp
 	return j
 }
 
+// NewJobForeground creates an amboy job that write all output,
```
Comment (Kimchelly) [inline job.go]: "writes"

File: job.go [hunk 4]
```diff
@@ -221,13 +253,22 @@ func amboyForegroundOutputJobFactory(pc ProcessConstructor) *amboyForegroundOutp
...
+// linwise to the current processes global grip logging instance with
```
Comment (Kimchelly) [inline job.go]: "linewise,"

File: job.go [hunk 5]
```diff
@@ -221,13 +253,22 @@ func amboyForegroundOutputJobFactory(pc ProcessConstructor) *amboyForegroundOutp
...
+// error and output sperpeated by level.
```
Comment (Kimchelly) [inline job.go]: "separated"

File: job.go [hunk 6]
```diff
@@ -221,13 +253,22 @@ func amboyForegroundOutputJobFactory(pc ProcessConstructor) *amboyForegroundOutp
...
+// processes in an existing Manager.
 func NewJobForeground(pc ProcessConstructor, opts *options.Create) Job {
 	j := amboyForegroundOutputJobFactory(pc)
 	j.SetID(fmt.Sprintf("%s.%x", j.Type().Name, opts.Hash()))
 	j.Options = opts
 	return j
 }
 
+// NewJobBasicForeground creates an amboy job that write all output,
+// linwise to the current processes global grip logging instance with
```
Comment (Kimchelly) [inline job.go]: "linewise,"

File: job.go [hunk 7]
```diff
@@ -221,13 +253,22 @@ func amboyForegroundOutputJobFactory(pc ProcessConstructor) *amboyForegroundOutp
...
+// Pass the process constructor to allow the amboy jobs to manipulate
+// processes in an existing Manager.
 func NewJobForeground(pc ProcessConstructor, opts *options.Create) Job {
 	j := amboyForegroundOutputJobFactory(pc)
 	j.SetID(fmt.Sprintf("%s.%x", j.Type().Name, opts.Hash()))
 	j.Options = opts
 	return j
 }
 
+// NewJobBasicForeground creates an amboy job that write all output,
```
Comment (Kimchelly) [inline job.go]: "writes"

File: job.go [hunk 8]
```diff
@@ -221,13 +253,22 @@ func amboyForegroundOutputJobFactory(pc ProcessConstructor) *amboyForegroundOutp
...
+// linwise to the current processes global grip logging instance with
+// error and output sperpeated by level.
```
Comment (Kimchelly) [inline job.go]: "separated"

File: subprocess_windows.go [hunk 9]
```diff
@@ -82,7 +82,8 @@ const (
 
 var (
 	modkernel32 = syscall.NewLazyDLL("kernel32.dll")
-	modadvapi32 = syscall.NewLazyDLL("advapi32.dll")
+	// commented the following unused mod
+	// modadvapi32 = syscall.NewLazyDLL("advapi32.dll")
```
Comment (Kimchelly) [inline subprocess_windows.go]: Can we just remove this? It's unused and I doubt it will ever be used.

File: subprocess_windows.go [hunk 10]
```diff
@@ -332,6 +333,7 @@ func WaitForSingleObject(object syscall.Handle, timeout time.Duration) (uint32,
 	return waitStatus, nil
 }
 
+// nolint
```
Comment (Kimchelly) [inline subprocess_windows.go]: We can just remove this function since it's old and unused.

File: tracker.go [hunk 11]
```diff
@@ -31,6 +31,7 @@ type mockProcessTracker struct {
 	Infos       []ProcessInfo
 }
 
+// nolint
```
Comment (Kimchelly) [inline tracker.go]: Why does this need nolint if it's used in linux tests?

File: harness.poplar (hunk 2 above)
Comment (tychoish) [reply harness.poplar]: I didn't want the linter to hit them before we migrage them to poplar, which is ticketed.

File: tracker.go (hunk 11 above)
Comment (tychoish) [reply tracker.go]: not sure
